zeppelin-zengine/src/main/java/org/apache/zeppelin/notebook/Paragraph.java

```diff
@@ -250,14 +250,18 @@ public void setInterpreter(Interpreter interpreter) {
         return note.getInterpreterCompletion();
       }
     }
+    this.interpreter = getBindedInterpreter();
+
     String trimmedBuffer = buffer != null ? buffer.trim() : null;
     cursor = calculateCursorPosition(buffer, trimmedBuffer, cursor);
 
+    setText(trimmedBuffer);
```

Comment:
should we change the text here?

Reply:
Yes, otherwise the cursor position may be incorrect in line 264 
```return this.interpreter.completion(this.scriptText, cursor, interpreterContext);```

Reply:
But why we trim the text, it is possible to have blank text in the front of paragraph. Changing text in the server side may not be a good user experience. Maybe there's better solution for cursor position.

Reply:
trim was added to fix autocomplete. For example autocomplete works incorrect (JDBC may contains name od datasource %jdbc (mysql)   ...)

Reply:
It seems jdbc specific issue, is it possible to fix it in jdbc interpreter instead of `Paragraph.java`.

Reply:
@zjffdu 
but you trim text in method setText()

```
this.scriptText = this.text.substring(headingSpace.length() + intpText.length() + 1).trim();
```

Reply:
@tinkoff-dwh Could you give one example that wrong cursor position is detected. I suspect it can be fixed in method `calculateCursorPosition`

Reply:
@zjffdu 
https://github.com/apache/zeppelin/blob/afd2bca46c778e9e84deb7ba5e4f5b258735022b/zeppelin-zengine/src/test/java/org/apache/zeppelin/notebook/ParagraphTest.java#L272

but what is the problem if the text in any case is trimmed? (in method setText)

Reply:
I suspect it would change the code of user write in frontend.

e.g.

```
jdbc

select * from table
```

may become to 
```
jdbc
select * from table
```

Changing user code is not user friendly experience IMO